logstash-core/src/main/java/co/elastic/logstash/api/Codec.java

```diff
@@ -43,28 +44,11 @@
     void flush(ByteBuffer buffer, Consumer<Map<String, Object>> eventConsumer);
 
     /**
-     * Encodes an {@link Event} and writes it into the specified {@link ByteBuffer}. Under ideal
-     * circumstances, the entirety of the event's encoding will fit into the supplied buffer. In cases
-     * where the buffer has insufficient space to hold the event's encoding, the buffer will be filled
-     * with as much of the event's encoding as possible, {@code false} will be returned, and the caller
-     * must call this method with the same event and a buffer that has more {@link Buffer#remaining()}
-     * bytes. That is typically done by draining the partial encoding from the supplied buffer. This
-     * process must be repeated until the event's entire encoding is written to the buffer at which
-     * point the method will return {@code true}. Attempting to call this method with a new event
-     * before the entirety of the previous event's encoding has been written to a buffer will result
-     * in an {@link EncodeException}.
-     *
-     * @param event  The event to encode.
-     * @param buffer The buffer into which the encoding of the event should be written. Codec
-     *               implementations are responsible for returning the buffer in a state from which it
-     *               can be read, typically by calling {@link Buffer#flip()} before returning.
-     * @return {@code true} if the entirety or final segment of the event's encoding was written to
-     * the buffer. {@code false} if the buffer was incapable of holding the entirety or remainder of the
-     * event's encoding.
-     * @throws EncodeException if called with a new event before the entirety of the previous event's
-     * encoding was written to a buffer.
+     * Encodes an {@link Event} and writes it to the specified {@link OutputStream}.
+     * @param event The event to encode.
+     * @param output The stream to which the encoded event should be written.
      */
-    boolean encode(Event event, ByteBuffer buffer) throws EncodeException;
+    void encode(Event event, OutputStream output) throws IOException;
```

Comment:
This is a breaking change to the API, which means that any plugins compiled against the current Beta API will fail ~with a runtime method lookup failure~ when validated by `PluginValidator`.

How are we calling this out to alleviate the pain of those who are already using the beta API?

Reply:
I can add something to the release notes and/or 7.1 release blog post explicitly calling that out. Would that be sufficient?

Reply:
I think so. Definitely release notes, but perhaps also release blog post.

I'm leaving this thread open until the PR for release notes is open (or a separate issue to track).

Reply:
@yaauie, I opened an issue here: https://github.com/elastic/logstash/issues/10751